share/new_eds/proofs_cache.go

```diff
@@ -29,8 +29,9 @@ var _ AccessorStreamer = (*proofsCache)(nil)
 type proofsCache struct {
 	inner AccessorStreamer
 
-	// lock protects axisCache
-	lock sync.RWMutex
+	// lock protects proofCache
+	lock       sync.RWMutex
```

Comment:
Why share the lock between multiple _independent_ fields, instead of using separate for each?

Reply:
Not really a big problem, because it is blasingly fast lock and very rare. It locks on setting pointer of AxisRoots, which will happen once

Reply:
I generally prefer to separate locks when possible. It is also better for readability.